cpu/sam0_common/periph/adc.c

```diff
@@ -87,66 +189,38 @@ static int _adc_configure(adc_res_t res)
         return -1;
     }
     _adc_poweroff();
-    if (ADC->CTRLA.reg & ADC_CTRLA_SWRST ||
-        ADC->CTRLA.reg & ADC_CTRLA_ENABLE ) {
+    if (ADC_DEV->CTRLA.reg & ADC_CTRLA_SWRST ||
+        ADC_DEV->CTRLA.reg & ADC_CTRLA_ENABLE ) {
         DEBUG("adc: not ready\n");
         return -1;
     }
-#ifdef CPU_SAMD21
-    /* Power On */
-    PM->APBCMASK.reg |= PM_APBCMASK_ADC;
-    /* GCLK Setup */
-    GCLK->CLKCTRL.reg = (uint32_t)(GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0 |
-                        (GCLK_CLKCTRL_ID(ADC_GCLK_ID)));
-    /* Configure CTRLB Register HERE IS THE RESOLUTION SET! */
-    ADC->CTRLB.reg = ADC_PRESCALER | res;
-    /* Load the fixed device calibration constants */
-    ADC->CALIB.reg =
-        ADC_CALIB_BIAS_CAL((*(uint32_t*)ADC_FUSES_BIASCAL_ADDR >>
-                            ADC_FUSES_BIASCAL_Pos)) |
-        ADC_CALIB_LINEARITY_CAL((*(uint64_t*)ADC_FUSES_LINEARITY_0_ADDR >>
-                                ADC_FUSES_LINEARITY_0_Pos));
+    _setup_clock();
+    _setup_calibration();
+    /* Set ADC resolution */
+#ifdef ADC_CTRLC_RESSEL
+    ADC_DEV->CTRLC.reg |= res;
+#else
+    ADC_DEV->CTRLB.reg |= res;
+#endif
     /* Set Voltage Reference */
-    ADC->REFCTRL.reg = ADC_REF_DEFAULT;
+    ADC_DEV->REFCTRL.reg = ADC_REF_DEFAULT;
     /* Disable all interrupts */
-    ADC->INTENCLR.reg = (ADC_INTENCLR_SYNCRDY) | (ADC_INTENCLR_WINMON) |
-                              (ADC_INTENCLR_OVERRUN) | (ADC_INTENCLR_RESRDY);
+    ADC_DEV->INTENCLR.reg = 0xFF;
+#ifdef CPU_SAMD21
     while (_adc_syncing()) {}
```

Comment:
You can move the `_adc_syncing()` out of the `ifdef` (or remove it entirely unless there is a reason why we have to sync before enabling the band gap. There is a sync at the end after enable, that should be enough.)

Reply:
I've moved it out of the ifdef. At least in the `ifdef` it was unnecessary to have it in both paths at the beginning.